tree/dataframe/src/RDFGraphUtils.cxx

```diff
@@ -61,13 +61,20 @@ std::string GraphCreatorHelper::RepresentGraph(RLoopManager *loopManager)
 {
 
    auto actions = loopManager->GetAllActions();
-   std::vector<std::shared_ptr<GraphNode>> leaves;
+   std::vector<std::shared_ptr<GraphNode>> nodes;
    for (auto action : actions) {
       // Triggers the graph construction. When action->GetGraph() will return, the node will be linked to all the branch
-      leaves.push_back(action->GetGraph());
+      nodes.emplace_back(action->GetGraph());
+   }
+
+   auto edges = loopManager->GetGraphEdges();
+   std::vector<std::shared_ptr<GraphNode>> leaves;
+   for (auto edge : edges) {
+      // Triggers the graph construction. When action->GetGraph() will return, the node will be linked to all the branch
+      nodes.emplace_back(edge->GetGraph());
```

Comment:
Or `std::transform` with a prior `nodes.resize(actions.size())` (which also prevents reallocs). But details.

Reply:
refactored to make it more readable and remove the extra allocations